fearless/Modules/Staking/SelectValidatorsFlow/SelectedValidatorList/SelectedValidatorListPresenter.swift

```diff
@@ -0,0 +1,109 @@
+import SoraFoundation
+
+final class SelectedValidatorListPresenter {
+    weak var view: SelectedValidatorListViewProtocol?
+    weak var delegate: SelectedValidatorListDelegate?
+
+    let wireframe: SelectedValidatorListWireframeProtocol
+    let viewModelFactory: SelectedValidatorListViewModelFactory
+    let maxTargets: Int
+
+    private var selectedValidatorList: [ElectedValidatorInfo]
+    private var viewModel: SelectedValidatorListViewModel?
+
+    init(
+        wireframe: SelectedValidatorListWireframeProtocol,
+        viewModelFactory: SelectedValidatorListViewModelFactory,
+        localizationManager: LocalizationManagerProtocol,
+        selectedValidators: [ElectedValidatorInfo],
+        maxTargets: Int
+    ) {
+        self.wireframe = wireframe
+        self.viewModelFactory = viewModelFactory
+        selectedValidatorList = selectedValidators
+        self.maxTargets = maxTargets
+        self.localizationManager = localizationManager
+    }
+
+    // MARK: - Private functions
+
+    private func createViewModel() -> SelectedValidatorListViewModel {
+        viewModelFactory.createViewModel(
+            from: selectedValidatorList,
+            totalValidatorsCount: maxTargets,
+            locale: selectedLocale
+        )
+    }
+
+    private func provideViewModel() {
+        let viewModel = createViewModel()
+        self.viewModel = viewModel
+        view?.reload(viewModel)
+    }
+
+    private func selectedValidatorInfo(from validator: ElectedValidatorInfo) -> SelectedValidatorInfo {
+        SelectedValidatorInfo(
+            address: validator.address,
+            identity: validator.identity,
+            stakeInfo: ValidatorStakeInfo(
+                nominators: validator.nominators,
+                totalStake: validator.totalStake,
+                stakeReturn: validator.stakeReturn,
+                maxNominatorsRewarded: validator.maxNominatorsRewarded
+            )
+        )
+    }
+}
+
+// MARK: - SelectedValidatorListPresenterProtocol
+
+extension SelectedValidatorListPresenter: SelectedValidatorListPresenterProtocol {
+    func setup() {
+        provideViewModel()
+    }
+
+    func didSelectValidator(at index: Int) {
+        let validatorInfo = selectedValidatorInfo(
+            from: selectedValidatorList[index]
+        )
+        wireframe.present(validatorInfo, from: view)
+    }
+
+    func removeItem(at index: Int) {
+        let validator = selectedValidatorList[index]
+        delegate?.didRemove(validator)
```

Comment:
Actually we haven't removed it yet :) may be call it before or after updating view